src/main/java/duke/model/planning/Recommendation.java

```diff
@@ -0,0 +1,65 @@
+package duke.model.planning;
+
+import duke.commons.exceptions.ApiException;
+import duke.commons.exceptions.ParseException;
+import duke.commons.exceptions.StartEndDateBeforeNowException;
+import duke.commons.exceptions.StartEndDateDiscordException;
+import duke.logic.api.ApiParser;
+import duke.logic.parsers.ParserTimeUtil;
+import duke.model.locations.Venue;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents an Itinerary and its contained information.
+ */
+public class Recommendation {
+    private List<Agenda> agendaList;
+
+    /**
+     * Constructor to initialise new Itinerary.
+     */
+    public Recommendation(List<Agenda> agendaList) {
+        this.agendaList = agendaList;
+    }
+
+    /**
+     * Returns number of days of the trip based on entered start and end dates.
+     *
+     * @param itineraryDetails contains all info to make an itinerary.
+     * @return The number of days of the trip
+     */
+
+    public Itinerary makeItinerary(String[] itineraryDetails) throws ParseException, ApiException,
+            StartEndDateBeforeNowException, StartEndDateDiscordException {
+        LocalDateTime start = ParserTimeUtil.parseStringToDate(itineraryDetails[1].strip());
+        LocalDateTime end = ParserTimeUtil.parseStringToDate(itineraryDetails[2].strip());
+        Venue hotelLocation = ApiParser.getLocationSearch(itineraryDetails[0].strip());
+
+        Itinerary itinerary = new Itinerary(start, end, hotelLocation, "New Recommendation");
+
+        List<Agenda> agendaList1 = new ArrayList<>(agendaList.subList(0,getNumberOfDays(start, end)));
+
+        assert (!agendaList1.isEmpty()) : "list should not be null";
+        itinerary.setTasks(agendaList1);
+
+        return itinerary;
+    }
+
+    private int getNumberOfDays(LocalDateTime start, LocalDateTime end) throws
```

Comment:
Catch and handle the API exception.
If it occurs, find some way to atleast have something functional.
Like make a new temporary hotelLocation or something. Don't just let the API exception be thrown all the way up.

Reply:
Yes this was one of the issues raised in the PE. I started by doing this in the recommend command. Will do it in the newItinerary soon